pkg/endpoint/endpoint.go

```diff
@@ -402,19 +409,31 @@ func (e *Endpoint) waitForProxyCompletions(proxyWaitGroup *completion.WaitGroup)
 
 // NewEndpointWithState creates a new endpoint useful for testing purposes
 func NewEndpointWithState(owner regeneration.Owner, proxy EndpointProxy, allocator cache.IdentityAllocator, ID uint16, state string) *Endpoint {
+	ep := createEndpoint(owner, proxy, allocator, ID, "")
+	ep.state = state
+	ep.eventQueue = eventqueue.NewEventQueueBuffered(fmt.Sprintf("endpoint-%d", ID), option.Config.EndpointQueueSize)
+
+	ep.UpdateLogger(nil)
+
+	ep.eventQueue.Run()
```

Comment:
Looping back around to answer my own question here, these are handled by `Expose()` for the host endpoint as well in this PR so there's nothing specific to this PR about the question. They will be initialized, it's just a bit of an unusual pattern.

Reply:
> I'm a bit surprised that we have a path that skips initializing these today and I'm a little lost as to why this initialization is spread across multiple places in the codebase

Yep, that wasn't the easiest to understand code when I started. Plus, I still don't understand the exact purpose of all the fields initialized here.

This PR isn't supposed to change the behaviors you describe though. `createEndpoint()` only does some code refactoring to avoid duplicate code and `CreateHostEndpoint()` should follow the general pattern of other endpoint creations.

Reply:
I agree with @joestringer that we should look at this outside of the context of this PR.

Reply:
> This PR isn't supposed to change the behaviors you describe though. createEndpoint() only does some code refactoring to avoid duplicate code and CreateHostEndpoint() should follow the general pattern of other endpoint creations.

I agree, as far as I can tell this is not changed by this PR.